core-blocks/block/edit-panel/index.js

```diff
@@ -22,9 +22,29 @@ class SharedBlockEditPanel extends Component {
 
 		this.titleField = createRef();
 		this.editButton = createRef();
+		this.editForm = createRef();
 		this.handleFormSubmit = this.handleFormSubmit.bind( this );
 		this.handleTitleChange = this.handleTitleChange.bind( this );
 		this.handleTitleKeyDown = this.handleTitleKeyDown.bind( this );
+		this.handleClickOutside = this.handleClickOutside.bind( this );
+	}
+
+	componentDidMount() {
+		document.addEventListener( 'click', this.handleClickOutside, true );
```

Comment:
I'd have considered suggesting [`withGlobalEvents`](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-global-events), though it doesn't support `useCapture`. Out of curiosity, why was that needed here?

Reply:
From the `react-click-outside` (which inspired this bit of code) readme: https://github.com/kentor/react-click-outside#details

> The enhanceWithClickOutside function wraps the provided component in another component that registers a click handler on document for the event capturing phase. Using the event capturing phase prevents elements with a click handler that calls stopPropagation from cancelling the click event that would eventually trigger the component's handleClickOutside function.